### ISSUE TITLE
Work with android.support.v4.app.DialogFragment

### ISSUE THREAD

**Issue by bagintz** [sjwall/MaterialTapTargetPrompt#97]
This does not work with the support library DialogFragment.  Is that possible?

**Comment by bagintz** [comment]
I was able to get it to "sort of" work by doing this from inside the android.support.v4.app.DialogFragment

`
        getDialog().setOwnerActivity(getActivity());
        new MaterialTapTargetPrompt.Builder(getDialog())
        .....
`

**Comment by sjwall** [comment]
Could you please describe the issue you are seeing in more detail? Testing a DialogFragment against sdk 27 shows a prompt contained within the DialogFragment bounds for me.

```java
public class DialogFragmentExample extends DialogFragment
{
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_bottom_sheet, container,
                false);
        getDialog().setTitle("DialogFragment");
        getDialog().setOnShowListener(new DialogInterface.OnShowListener()
        {
            @Override
            public void onShow(DialogInterface dialog)
            {
                new MaterialTapTargetPrompt.Builder(getDialog())
                        .setPrimaryText(R.string.search_prompt_title)
                        .setSecondaryText(R.string.search_prompt_description)
                        .setAnimationInterpolator(new FastOutSlowInInterpolator())
                        .setMaxTextWidth(R.dimen.tap_target_menu_max_width)
                        .setIcon(R.drawable.ic_search)
                        .setTarget(R.id.bs_search)
                        .show();
            }
        });
        return rootView;
    }
}
```

**Comment by bagintz** [comment]
checking on it, will let you know

**Comment by bagintz** [comment]
@sjwall is your DialogFragment from above a support DialogFragment? (android.support.v4.app.DialogFragment)

**Comment by bagintz** [comment]
I can get it to work if I put `getDialog().setOwnerActivity(getActivity());` above the new Material..., but then it stretches the dialog fragment to 100% height, which doesn't work :(

**Comment by sjwall** [comment]
It is because the prompt is being added to the dialog view hierarchy instead of the view containing it. I believe that using a PopupWindow would fix it and I will look at the effect on changing the library to use a PopupWindow. It might work to extend `DialogResourceFinder`, override `getPromptParentView` and return the decorview parent view. I haven't had chance to test to see if that works though

**Comment by sjwall** [comment]
Constructors have been added to the `Builder` class for support library fragments and dialog fragments in PR #114 and released in `v2.8.0`
